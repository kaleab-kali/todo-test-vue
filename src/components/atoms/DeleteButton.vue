<template>
  <button :class="$style.delete" data-cy="delete" @click="$emit('click')">
    &times;
  </button>
</template>

<script>
export default {};
</script>

<style lang="scss" module>
.delete {
  margin-right: 1.25rem;
  border: 1.5px solid $lightOrange;
  border-radius: 2px;
  color: $lightOrange;
  background-color: $darkPeach;
  max-height: 1.5rem;
}
.delete:active {
  color: red;
}
.delete:hover {
  border: 1.5px solid $orange;
  color: $orange;
  cursor: pointer;
}
</style>
