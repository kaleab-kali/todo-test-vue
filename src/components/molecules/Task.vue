<template>
  <div>
    <div :class="$style.todo">
      <Checkbox @click="$emit('changes')" :isChecked="isChecked" />
      <span
        :class="[$style.task, { [$style.done]: isChecked }]"
        data-cy="task__title"
        >{{ title }}</span
      >
      <DeleteButton @click="$emit('delete')" />
    </div>
  </div>
</template>

<script>
import Checkbox from "@/components/atoms/CheckBox";
import DeleteButton from "@/components/atoms/DeleteButton";
export default {
  props: {
    title: String,
    isChecked: Boolean,
  },
  components: {
    Checkbox,
    DeleteButton,
  },
};
</script>

<style lang="scss" module>
.todo {
  font-family: "Inter";
  background-color: $darkPeach;
  max-width: 30rem;
  margin-bottom: 1.56rem;
  display: flex;
  color: $brown;
  font-weight: 400;
  font-size: 1.25rem;
  padding: 0.688rem 0;
  margin-top: 0;
  border-radius: 0.625rem;
  .done {
    text-decoration: line-through;
  }
  @media (max-width: 980px) {
    max-width: 30rem;
    margin: 1rem;
  }
  @media (max-width: 850px) {
    font-size: 1rem;
  }
  @media (max-width: 600px) {
    font-size: 0.8rem;
  }
  .task {
    margin-left: -8rem;
    position: relative;
    @media (max-width: 850px) {
      input {
        font-size: 1rem;
      }
    }
    @media (max-width: 600px) {
      margin-left: -6rem;
      input {
        max-width: 10rem;
        font-size: 0.9rem;
      }
    }
    @media (max-width: 450px) {
      margin-left: -3rem;
    }
  }
}
.todo:last-child {
  margin-bottom: 0.56rem;
}
</style>
