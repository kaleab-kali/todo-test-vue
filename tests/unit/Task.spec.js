import { mount } from "@vue/test-utils";
import Task from "@/components/molecules/Task.vue";
import Checkbox from "@/components/atoms/Checkbox";
import DeleteButton from "@/components/atoms/DeleteButton";

describe("Molecule Task Component", () => {
  it("Correct unchecked layout", () => {
    const wrapper = mount(Task, {
      propsData: {
        title: "Test Title",
        isChecked: false,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("Correct checked layout", () => {
    const wrapper = mount(Task, {
      propsData: {
        title: "Test Title",
        isChecked: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("Title renders correctly", () => {
    const title = "Test Title";
    const wrapper = mount(Task, {
      propsData: {
        title,
        isChecked: true,
      },
    });
    expect(wrapper.find("span.task").text()).toBe(title);
  });
  it("Checkbox is checked", () => {
    const wrapper = mount(Task, {
      propsData: {
        isChecked: true,
      },
    });
    expect(wrapper.findComponent(Checkbox).props().isChecked).toBe(true);
  });
  it("All Checkbox is checked", async () => {
    const wrapper = mount(Task, {
      propsData: {
        isChecked: true,
      },
    });
    await wrapper.findComponent(Checkbox).trigger("click");
    await wrapper.findComponent(DeleteButton).trigger("click");
    expect(wrapper.emitted().delete).toBeTruthy();
    expect(wrapper.emitted().changes).toBeTruthy();
  });
});
