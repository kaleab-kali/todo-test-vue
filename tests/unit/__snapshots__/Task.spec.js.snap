// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule Task Component Correct checked layout 1`] = `
<div
  class="todo"
>
  <label>
    <input
      class="checkbox"
      type="checkbox"
    />
     
    <span
      class="fake"
    />
  </label>
   
  <span
    class="task"
  >
    Test Title
  </span>
   
  <button
    class="delete"
  >
    ×
  </button>
</div>
`;

exports[`Molecule Task Component Correct layout 1`] = `
<div
  class="todo"
>
  <label>
    <input
      class="checkbox"
      type="checkbox"
    />
     
    <span
      class="fake"
    />
  </label>
   
  <span
    class="task"
  >
    Test Title
  </span>
   
  <button
    class="delete"
  >
    ×
  </button>
</div>
`;

exports[`Molecule Task Component Correct unchecked layout 1`] = `
<div
  class="todo"
>
  <label>
    <input
      class="checkbox"
      type="checkbox"
    />
     
    <span
      class="fake"
    />
  </label>
   
  <span
    class="task"
  >
    Test Title
  </span>
   
  <button
    class="delete"
  >
    ×
  </button>
</div>
`;
